<!DOCTYPE html>
<html>
    <head>
        <meta charset="UTF-8">
        <link href="https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap" rel="stylesheet"> 
        <link href="/static/main.css" rel="stylesheet" type="text/css">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <title>Web uploader for Public SARS-CoV-2 Sequence Resource</title>
    </head>
    <body>
      <section class="header">
      <div class="logo"><a href="http://covid-19.genenetwork.org/"><img src="static/image/coronasmallcomp.gif" width="150" title="COVID-19 image by Tyler Morgan-Wall"></a></div>
      <h1>Web uploader for Public SARS-CoV-2 Sequence Resource</h1>

<small>Disabled until we got everything wired up</small>

      </section>

      <section class="search-section">
          <div class="filter-options" action="#">
              <p>[Demo] Display content sequences by: </p>

              <div>
                  <button class="button" onclick="fetchSEQBySpecimen()">Specimen Source</button>
                  <button class="button" onclick="fetchSEQByLocation()">Location</button>
                  <button class="button" onclick="fetchSEQByTech()">Tech</button>
                  <button class="button" onclick="fetchAllaccessions()">Allaccessions</button>
              </div>

          </div>

          <div class="search">
              <input id="search-input" id="global-search" type="search" placeholder="FASTA uri" required>
              <button class="button search-button" type="submit" onclick="search()">
                  <span class="icon ion-search">
                      <span class="sr-only">Search</span>
                  </span>
              </button>
          </div>
      </section>

      <div id="loader" class="loader invisible"></div>

      <section id="results" class="invisible">
          <pre><code id="json"></code></pre>
      </section>

        <section>
            <form action="/submit" method="POST" enctype="multipart/form-data" id="main_form" class="grid-container">
                <div class="intro">
                    <p>
                        Upload your SARS-CoV-2 sequence (FASTA or FASTQ formats) with metadata (JSONLD) to the <a href="https://workbench.lugli.arvadosapi.com/collections/lugli-4zz18-z513nlpqm03hpca">public sequence resource</a>. The upload will trigger a
                        recompute with all available sequences into a Pangenome
                        available for
                        <a href="https://workbench.lugli.arvadosapi.com/collections/lugli-4zz18-z513nlpqm03hpca">download</a>!
                    </p>
                    <p>
                        Your uploaded sequence will automatically be processed
                        and incorporated into the public pangenome with
                        metadata using worklows from the High Performance Open Biology Lab defined <a href="https://github.com/hpobio-lab/viral-analysis/tree/master/cwl/pangenome-generate">here</a>. All data is published under
                        a <a href="https://creativecommons.org/licenses/by/4.0/">Creative
                        Commons 4.0 attribution license</a> (CC-BY-4.0). You
                        can take the published (GFA/RDF/FASTA) data and store it in
                        a triple store for further processing. We also plan to
                        combine identifiers with clinical data stored securely at <a href="https://redcap-covid19.elixir-luxembourg.org/redcap/">REDCap</a>.
                        A free command line version of the uploader can be
                        installed from <a href="https://github.com/arvados/bh20-seq-resource">source</a>.
                    </p>

                </div>

                <div class="fasta-file-select">
                    <h2><svg class="bi bi-cloud-upload" width="1.2em" height="1.2em" viewBox="0 0 16 16" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
                        <path d="M4.887 6.2l-.964-.165A2.5 2.5 0 103.5 11H6v1H3.5a3.5 3.5 0 11.59-6.95 5.002 5.002 0 119.804 1.98A2.501 2.501 0 0113.5 12H10v-1h3.5a1.5 1.5 0 00.237-2.981L12.7 7.854l.216-1.028a4 4 0 10-7.843-1.587l-.185.96z"/>
                        <path fill-rule="evenodd" d="M5 8.854a.5.5 0 00.707 0L8 6.56l2.293 2.293A.5.5 0 1011 8.146L8.354 5.5a.5.5 0 00-.708 0L5 8.146a.5.5 0 000 .708z" clip-rule="evenodd"/>
                        <path fill-rule="evenodd" d="M8 6a.5.5 0 01.5.5v8a.5.5 0 01-1 0v-8A.5.5 0 018 6z" clip-rule="evenodd"/>
                    </svg> Upload SARS-CoV-2 Sequence</h2>
                    
                    <label for="fasta">Select FASTA file of assembled genome (max 50K), or FASTQ of reads (<span class="dropt" title="For a larger fastq file you'll need to use a CLI uploader">max 150MB<span style="width:500px;"></span></span>) : </label>
                    <br>
                    <input type="file" id="fasta" name="fasta" accept=".fa,.fasta,.fna,.fq" required>
                    <br>
                    <small>
                    Note that by uploading your data you automatically agree to a <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY-4.0 license</a>. </small>
                </div>

                <div class="metadata">
                    <label>Select metadata submission method:</label>
                    <br>
                    <input type="radio" id="metadata_form" name="metadata_type" value="fill" checked  onchange="setMode()" required>
                    <label for="metadata_form">Fill in metadata manually</label>
                    <input type="radio" id="metadata_upload" name="metadata_type" value="upload" onchange="setMode()" required>
                    <label for="metadata_upload">Upload metadata file</label>
                    <br>
                    <small>Make sure the metadata has submitter attribution details.</small>

                    <div id="metadata_upload_form_spot">
                        <div id="metadata_upload_form">
                          <br>
                          <label for="metadata">Select JSON or YAML metadata file following <a href="https://github.com/arvados/bh20-seq-resource/blob/master/bh20sequploader/bh20seq-schema.yml" target="_blank">this schema</a> and <a href="https://github.com/arvados/bh20-seq-resource/blob/master/example/metadata.yaml" target="_blank">example</a> (max 50K):</label>
                          <br>
                          <input type="file" id="metadata" name="metadata" accept=".json,.yml,.yaml" required>
                          <br>
                        </div>
                    </div>

                </div>
                <div id="metadata_fill_form_spot">
                    <div id="metadata_fill_form">
                        {% for record in fields %}

                        {% if 'heading' in record %}
                        {% if loop.index > 1 and 2 < 3  %}
                    </div>
                    {% endif %}
                    <div class="record">
                        <h4>{{ record['heading'] }}</h4>
                        {% else %}
                        <div class="field-group" data-keypath="{{ record['id'] }}">
                            <div class="field" data-number="0">
                                <label for="{{ record['id'] }}{{ '[0]' if record['list'] else ''}}" title="{{ record.get('docstring', '') }}">
                                    {{ record['label'] }}
                                    {{ "*" if record['required'] else "" }}
                                    {% if 'docstring' in record %}
                                    <a href='javascript:alert({{ record['docstring'] | tojson }})'>‚ùì</a>
                                    {% endif %}
                                    {% if 'ref_iri' in record %}
                                    <a href="{{ record['ref_iri'] }}" target="_blank" title="Ontology Link">üîó</a>
                                    {% endif %}
                                </label>
                                {% if record['type'] == 'select' %}
                                <select class="control" id="{{ record['id'] }}{{ '[0]' if record['list'] else ''}}" name="{{ record['id'] }}{{ '[0]' if record['list'] else ''}}" {{ "required" if record['required'] else "" }}>
                                    <option value="" selected>Choose one...</option>
                                    {% for option in record['options'] %}
                                    <option value="{{ option[1] }}">{{ option[0] }}</option>
                                    {% endfor %}
                                </select>
                                {% else %}
                                <input class="control" type="{{ record['type'] }}" id="{{ record['id'] }}{{ '[0]' if record['list'] else ''}}" name="{{ record['id'] }}{{ '[0]' if record['list'] else ''}}" {{ "required" if record['required'] else "" }} {{ ("step=" + record['step']) if 'step' in record else ""}}>
                                {% endif %}
                            </div>
                            {% if record['list'] %}
                            <button type="button" title="Remove field" class="remove-field invisible">‚ûñ</button>
                            <button type="button" title="Add field" class="add-field">‚ûï</button>
                            {% endif %}
                        </div>
                        {% endif %}
                        {% if loop.index == loop.length %}
                    </div>
                    {% endif %}
                    {% endfor %}
                </div>

                </div>

                <input class="submit" type="submit" value="Add to Pangenome">
            </form>
        </section>
<br>
<div class="about">
    <div>
        <h1>ABOUT</h1>
        <p>
            This a public repository created at the COVID-19 BioHackathon
            that has a low barrier to entry for uploading sequence data using
            best practices. I.e., data is published with a creative commons
            4.0 (CC-4.0) license with metadata using state-of-the art
            standards and, perhaps most importantly, providing standardized
            workflows that get triggered on upload, so that results are
            immediately available in standardized data formats. The repository
            will be maintained and expanded for the duration of the
            pandemic. To contribute data simply upload it! To contribute code
            and/or workflows see
            the <a href="https://github.com/arvados/bh20-seq-resource">project
            repository</a>. For more information see the <a href="https://github.com/arvados/bh20-seq-resource/blob/master/paper/paper.md">paper</a> (WIP).
        </p>

    </div>
  <div class="sponsors">
      <a href="https://arvados.org/"><img src="static/image/arvados-logo.png"></a>
      <a href="https://www.commonwl.org/"><img src="static/image/CWL-Logo-Header.png"></a>

      <a href="https://github.com/virtual-biohackathons/covid-19-bh20">
          <img src="static/image/covid19biohackathon.png"></a>
  </div>
</div>
<div class="footer">
    <!-- Sponsors -->


        <center>
        <small><a href="https://github.com/arvados/bh20-seq-resource">Source code</a> &middot; Powered by <a href="https://www.commonwl.org/">Common Workflow Language</a> &amp; <a href="https://arvados.org/">Arvados</a>; Made for <a href="https://github.com/virtual-biohackathons/covid-19-bh20">COVID-19-BH20</a>
        </small>
        </center>


</div>

<script type="text/javascript">
    let scriptRoot = {{ request.script_root|tojson|safe }};
</script>

<script type="text/javascript" src="/static/main.js"></script>
    </body>
</html>
